### PR TITLE
docs(components): [statistic]

### PR DESCRIPTION
**Pull request by chenxch**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/statistic/src/countdown.ts [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
   format: {
     type: String,
     default: 'HH:mm:ss',
   },
+  /**
+   * @description Sets the prefix of a countdown
+   */
   prefix: {
     type: String,
-    default: '',
```
Comment (holazz) [inline packages/components/statistic/src/countdown.ts:14]: **Suggestion**
```ts
prefix: {
  type: String,
}
```
to
```ts
prefix: String
```

Same in other places.

File: packages/components/statistic/src/countdown.ts [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+  /**
+   * @description countdown titles
+   */
   title: {
-    type: [String, Number],
-    default: '',
+    type: String,
   },
+  /**
+   * @description Styles countdown values
```
Comment (holazz) [inline packages/components/statistic/src/countdown.ts:33]: The description here seems to be misplaced.

File: packages/components/statistic/src/countdown.ts [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+   */
   value: {
     type: definePropType<number | Dayjs>([Number, Object]),
   },
+  /**
+   * @description numerical precision
```
Comment (holazz) [inline packages/components/statistic/src/countdown.ts:39]: Also here.

In addition, I suggest that the type here be modified to:

```diff
valueStyle: {
-  type: definePropType<StyleValue>([String, Object]),
-  default: () => ({}),
+  type: definePropType<StyleValue>([String, Object, Array]),
}
```

File: packages/components/statistic/src/countdown.ts [hunk 4]
```diff
@@ -35,12 +49,8 @@ export const countdownEmits = {
   change: (value: number) => !!value,
 }
 
-export const formatTimeStr = function (
-  format: any,
-  time: number,
-  futureTime: any
-) {
-  const timeUnits: any = [
+export const formatTimeStr = function (format: string, time: number) {
```
Comment (holazz) [inline packages/components/statistic/src/countdown.ts:52]: **Suggestion**:

```ts
export const formatTimeStr = (format: string, time: number) => {}
```
or
```ts
export function formatTimeStr(format: string, time: number) => {}
```

Same in other places.

File: packages/components/statistic/src/statistic.ts [hunk 5]
```diff
@@ -2,47 +2,71 @@ import { buildProps, definePropType } from '@element-plus/utils'
...
   },
+  /**
+   * @description Custom numerical presentation
+   */
   formatter: {
     type: definePropType<(val: string | number) => string | number>(Function),
   },
+  /**
+   * @description Numerical content
+   */
```
Comment (holazz) [inline packages/components/statistic/src/statistic.ts:34]: `default: null` at line#37 seems unnecessary.

File: docs/en-US/component/statistic.md [hunk 6]
```diff
@@ -42,15 +42,14 @@ statistic/card
 | Attribute         | Description                    | Type                                                                | Default |
 | ----------------- | ------------------------------ | ------------------------------------------------------------------- | ------- |
 | value             | Numerical content              | ^[string] / ^[number]                                               | —       |
-| decimal-separator | Setting the decimal point      | ^[string]                                                           | —       |
-| format            | Custom numerical presentation  | ^[string]                                                           | —       |
+| decimal-separator | Setting the decimal point      | ^[string]                                                           | .       |
+| formatter         | Custom numerical presentation  | ^[string]                                                           | —       |
```
Comment (holazz) [inline docs/en-US/component/statistic.md:46]: `Function` type is missing here.

**Review by holazz — COMMENTED** [review]
I left a few comments for your consideration.😊

File: packages/components/statistic/src/statistic.vue [hunk 7]
```diff
@@ -53,12 +53,15 @@ const disposeValue = computed(() => {
         .slice(1)}`
       decimal = decimal.slice(0, props.precision)
     }
-    integer = regroup(integer, props.rate, props.groupSeparator)
+    integer = groupFormat(integer, 3, props.groupSeparator)
```
Comment (ryuhangyeong) [inline packages/components/statistic/src/statistic.vue:56]: ## Suggestion

How about extracting it as a variable to clarify the meaning of 3?

File: packages/components/statistic/src/countdown.ts [hunk 8]
```diff
@@ -35,12 +49,8 @@ export const countdownEmits = {
   change: (value: number) => !!value,
```
Comment (ryuhangyeong) [inline packages/components/statistic/src/countdown.ts:49]: ## Suggestion

```
import { CHANGE_EVENT } from '@element-plus/constants'
import {
  isNumber,
} from '@element-plus/utils'

export const countdownEmits = {
  finish: () => true,
  [CHANGE_EVENT ]: (value: number) => isNumber(value),
}
```

File: docs/en-US/component/statistic.md [hunk 9]
```diff
@@ -63,26 +62,29 @@ statistic/card
...
 
 ### Countdown Events
 
-| Method | Description                                | Type                                     |
-| ------ | ------------------------------------------ | ---------------------------------------- |
-| change | Enable in the 'countdown' function         | ^[Function]`() => Date`                  |
-| finish | Launched after the 'countdown' is complete | ^[Function]`(event: FocusEvent) => void` |
+| Method | Description                  | Type                                 |
+| ------ | ---------------------------- | ------------------------------------ |
+| change | Time difference change event | ^[Function]`(value: number) => Date` |
```
Comment (ryuhangyeong) [inline docs/en-US/component/statistic.md:86]: ## Question

Is the return type Date correct?

File: packages/components/statistic/src/countdown.ts (hunk 1 above)
Comment (chenxch) [reply packages/components/statistic/src/countdown.ts:14]: ok

File: packages/components/statistic/src/countdown.ts (hunk 2 above)
Comment (chenxch) [reply packages/components/statistic/src/countdown.ts:33]: thx~

File: packages/components/statistic/src/countdown.ts (hunk 3 above)
Comment (chenxch) [reply packages/components/statistic/src/countdown.ts:39]: ok

File: packages/components/statistic/src/statistic.vue (hunk 7 above)
Comment (chenxch) [reply packages/components/statistic/src/statistic.vue:56]: You're right.

File: packages/components/statistic/src/countdown.ts (hunk 8 above)
Comment (chenxch) [reply packages/components/statistic/src/countdown.ts:49]: ok

File: docs/examples/statistic/card.vue [hunk 10]
```diff
@@ -30,7 +26,7 @@
               <el-row>
                 <el-col :span="12" :xs="24">
                   <span class="item"
-                    >比昨日
+                    >比昨天
```
Comment (kooriookami) [inline docs/examples/statistic/card.vue:29]: Please translate.

File: docs/examples/statistic/card.vue [hunk 11]
```diff
@@ -40,7 +36,7 @@
                 ></el-col>
                 <el-col :span="12" :xs="24">
                   <span class="item" style="text-align: right"
-                    >比前日
+                    >比前天
```
Comment (kooriookami) [inline docs/examples/statistic/card.vue:39]: Please translate.

File: docs/examples/statistic/card.vue [hunk 12]
```diff
@@ -127,15 +115,16 @@
             <div class="s-Bottom">
               <el-row>
                 <el-col :span="22">
-                  <span class="item"
-                    >比昨日
+                  <span class="item">
+                    比昨天
```
Comment (kooriookami) [inline docs/examples/statistic/card.vue:119]: Please translate.

File: packages/components/statistic/__tests__/countdown.test.tsx [hunk 13]
```diff
@@ -1,6 +1,11 @@
 import { mount } from '@vue/test-utils'
 import { describe, expect, it } from 'vitest'
+import dayjs from 'dayjs'
 import Countdown from '../src/countdown.vue'
+
+const wait = (ms = 100) =>
+  new Promise((resolve) => setTimeout(() => resolve(0), ms))
+
```
Comment (kooriookami) [inline packages/components/statistic/__tests__/countdown.test.tsx:8]: There's a ready-made function in `packages/test-utils/sleep.ts`.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/statistic/src/countdown.ts [hunk 14]
```diff
@@ -1,46 +1,50 @@
...
 
-export const formatTimeStr = function (
-  format: any,
-  time: number,
-  futureTime: any
-) {
-  const timeUnits: any = [
+export const formatTimeStr = (format: string, time: number) => {
+  const timeUnits: [string, number][] = [
     ['Y', 1000 * 60 * 60 * 24 * 365], // years
```
[the remark this answers is not in the document]
Comment (chenxch) [reply packages/components/statistic/src/countdown.ts:48]: Don't consider it, just do regular processing, if it doesn't meet the requirements, it is recommended to customize the format.

File: packages/components/statistic/__tests__/countdown.test.tsx (hunk 13 above)
Comment (chenxch) [reply packages/components/statistic/__tests__/countdown.test.tsx]: adjusted.

**Review by kooriookami — APPROVED** [review]

File: docs/examples/statistic/countdown.vue [hunk 15]
```diff
@@ -3,28 +3,28 @@
...
+            <el-button type="primary" @click="reset">Reset</el-button>
           </div>
         </el-col>
         <el-col :span="8">
-          <ElCountdown
-            :value="value4"
-            countdown
-            format="DD 天 HH 时 mm 分 ss 秒"
-          >
+          <ElCountdown :value="value2" format="DD HH : mm : ss ">
```
Comment (kooriookami) [inline docs/examples/statistic/countdown.vue:19]:
```suggestion
          <ElCountdown :value="value2" format="DD Days HH:mm:ss ">
```
No need to use space between date.

**Comment by kooriookami** [comment]
image

**Pull request merged**